### PR TITLE
Feature/storing workspace id in session

### PR DESCRIPTION
**Pull request by arpita-20**

### CODE REVIEW

File: app/models/tests.server.ts [hunk 1]
```diff
@@ -17,12 +17,13 @@ export async function getTestById({ id }: Pick<Section, 'id'>) {
   })
 }
 
-export async function getAllTests(obj: any) {
+export async function getAllTests(obj: any, currentWorkspaceId: string) {
```
Comment (richa-copods) [inline app/models/tests.server.ts:20]: Please give proper type

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/models/tests.server.ts (hunk 1 above)
Comment (arpita-20) [reply app/models/tests.server.ts:20]: it is the same bug we are getting earlier that's why not changed the type

File: app/components/side-nav/SideNav.tsx [hunk 2]
```diff
@@ -71,11 +74,29 @@ const sideNavGuide = [
 const SideNav = () => {
   const { t } = useTranslation()
 
+  const { workspaces = [], currentWorkspaceId } = useLoaderData()
+
+  const tempWorkspaces = workspaces.map((wk: UserWorkspace) => {
```
Comment (anuragP04) [inline app/components/side-nav/SideNav.tsx:79]: wk?? use some verbose variable names

File: app/models/tests.server.ts (hunk 1 above)
Comment (anuragP04) [reply app/models/tests.server.ts:20]: what is obj here, give some relevant name.....and try to solve any error that is occurring, try avoiding `any` type

File: app/routes/dashboard.tsx [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
 import type { LoaderFunction } from '@remix-run/node'
 import { routes } from '~/constants/route.constants'
+import { getUserWorkspaces } from '~/models/workspace.server'
 
 export const loader: LoaderFunction = async ({ request }) => {
   const userId = await getUserId(request)
+  const currentWorkspaceId = await getWorkspaceId(request)
   if (!userId) return redirect(routes.signIn)
-  return null
+  const workspaces = await getUserWorkspaces(userId)
```
Comment (anuragP04) [inline app/routes/dashboard.tsx:12]: why workspace is imported here, we don't have dashboard page I think

File: app/models/result.server.ts [hunk 4]
```diff
@@ -190,16 +197,16 @@ export async function updateCandidateStatus({
   })
 }
 
-export async function getAllCandidateTests(obj: object) {
+export async function getAllCandidateTests(obj: object, workspaceId: string) {
```
Comment (anuragP04) [inline app/models/result.server.ts:200]: type object?

File: app/models/sections.server.ts [hunk 5]
```diff
@@ -24,12 +24,13 @@ export async function getSectionById({ id }: Pick<Section, 'id'>) {
   })
 }
 
-export async function getAllSections(obj: string) {
+export async function getAllSections(obj: string, currentWorkspaceId: string) {
```
Comment (anuragP04) [inline app/models/sections.server.ts:27]: maintain consistency in naming arguments.....some places you have used `workspaceId` and on some places `currentWorkspaceId`

File: app/routes/dashboard.tsx (hunk 3 above)
Comment (arpita-20) [reply app/routes/dashboard.tsx:12]: yes we don't have dashboard page.. but when user login they redirected to dashboard page first

File: app/session.server.ts [hunk 6]
```diff
@@ -32,6 +33,19 @@ export async function getUserId(
...
+  request: Request
+): Promise<string | undefined> {
+  const session = await getSession(request)
+  const workspaceId = session.get(USER_WORKSPACE_KEY)
+  return workspaceId
+}
+
+export async function getDefaultWorkspaceIdForUser(userId: string) {
+  const workspaceId = await getDefaultWorkspaceIdForUserQuery(userId as string)
+  return workspaceId?.workspace[0]?.id
```
Comment (richa-copods) [inline app/session.server.ts:46]: Please explain the logic for doing this

File: prisma/migrations/20221007092832_workspace_queries/migration.sql [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+DO $$
+DECLARE 
+  workspaceId varchar;
+  userWorkspaceId varchar;
+  userList record;
+BEGIN
+    FOR userList in SELECT *  FROM "User" LOOP
+      workspaceId=gen_random_uuid();
+      userWorkspaceId=gen_random_uuid();  
+      PERFORM * FROM "UserWorkspace" where "userId"=userList."id" AND "isDefault"=NULL;
```
Comment (richa-copods) [inline prisma/migrations/20221007092832_workspace_queries/migration.sql:12]: Please explain the logic of doing `"isDefault"=NULL` as in Schema we are giving `false` as default value

File: app/models/sections.server.ts [hunk 8]
```diff
@@ -24,12 +24,16 @@ export async function getSectionById({ id }: Pick<Section, 'id'>) {
   })
 }
 
-export async function getAllSections(obj: string) {
-  let filter = obj ? obj : '{"orderBy":{"createdAt":"asc"}}'
+export async function getAllSections(
+  filterData: string,
+  currentWorkspaceId: string
```
Comment (richa-copods) [inline app/models/sections.server.ts:29]: Please maintain consistency in naming of function parameters. Here it's `currentWorkspaceId` and at other places it's `workspaceId`

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/models/tests.server.ts [hunk 9]
```diff
@@ -17,12 +17,15 @@ export async function getTestById({ id }: Pick<Section, 'id'>) {
   })
 }
 
-export async function getAllTests(obj: any) {
-  let filter = obj.orderBy ? obj : { orderBy: { name: sortByOrder.ascending } }
+export async function getAllTests(filterData: any, workspaceId: string) {
```
[the remark this answers is not in the document]
Comment (arpita-20) [reply app/models/tests.server.ts:20]: as told earlier, we were getting stuck here previously, that's why added any here

File: app/session.server.ts (hunk 6 above)
Comment (arpita-20) [reply app/session.server.ts:46]:
```
export async function getDefaultWorkspaceIdForUserQuery(userId: string) {
  return prisma.user.findUnique({
    where: { id: userId },
    select: {
      workspace: {
        select: {
          id: true,
        },
      },
    },
  })
}
```

we are adding this function which is returning array as output and we want id of the workspace for every user

File: app/components/side-nav/SideNav.tsx [hunk 10]
```diff
@@ -3,6 +3,9 @@ import Header from '~/components/SideNavHeader'
 import Footer from '~/components/SideNavFooter'
 import { routes } from '~/constants/route.constants'
 import { useTranslation } from 'react-i18next'
+import DropdownField from '../form/Dropdown'
+import { useLoaderData } from '@remix-run/react'
+import type { UserWorkspace } from '~/interface/Interface'
 
 const sideNavGuide = [
   // {
```
[the remark this answers is not in the document]
Comment (arpita-20) [reply app/components/side-nav/SideNav.tsx:11]: those commented code will be used in future.. that's why it is there

File: app/session.server.ts [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import { createCookieSessionStorage, redirect } from '@remix-run/node'
 import invariant from 'tiny-invariant'
 
-import type { User } from '~/models/user.server'
+import { User, getDefaultWorkspaceIdForUserQuery } from '~/models/user.server'
```
Comment (richa-copods) [inline app/session.server.ts:4]: Please get rid of this warning

File: prisma/migrations/20221007092832_workspace_queries/migration.sql (hunk 7 above)
Comment (richa-copods) [reply prisma/migrations/20221007092832_workspace_queries/migration.sql]: Please check everything is working as expected....

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: prisma/migrations/20221007092832_workspace_queries/migration.sql (hunk 7 above)
Comment (arpita-20) [reply prisma/migrations/20221007092832_workspace_queries/migration.sql]: yes it is working.. i have checked it...

**Review by richa-copods — APPROVED** [review]

**Pull request merged**